ui/src/message_popup/renderers/instant_audio_recorder.jsx

```diff
@@ -0,0 +1,144 @@
+/* @flow weak */
+import React from 'react';
+import RaisedButton from 'material-ui/RaisedButton';
+import AudioCapture from '../../components/audio_capture.jsx';
+
+type State = {
+  haveRecorded: bool,
+  downloadUrl: ?string,
+  uploadState: 'idle' | 'pending' | 'done',
+  blob: ?Blob,
+  uploadedUrl: ?string
+};
+
+export default React.createClass({
```

Comment:
If I read this right, this is the same as [AudioRecorderFlow](https://github.com/mit-teaching-systems-lab/threeflows/blob/master/ui/src/components/audio_recorder_flow.jsx), but with some steps in the process missing, is that right?

It'd be good to call out the difference and compare them in a comment at the top of the class here.  There might be opportunities to generalize, but it seems like the essence of these two components are their transitions between the different states so also fine to keep them separate.